dist-raw/node-cjs-helpers.js

```diff
@@ -1,4 +1,4 @@
-const {ArrayPrototypeForEach, StringPrototypeStartsWith, ObjectPrototypeHasOwnProperty, StringPrototypeIncludes, ObjectDefineProperty} = require('./node-primordials');
```

Comment:
Is this change safe to make on every node version we support?  Is it necessary?

Reply:
We can keep expanding our own version of `node-primordials`, but why it's not using upstream file as it instead of cherry picking individual methods a better option? That is a lot more work in terms of maintenance and also changes in node primordials may end up diverging from our file.

Reply:
From what I understand, the node primordials will never change in complex ways, because they are all core JavaScript methods and are used throughout node's codebase.  So the maintenance work has actually been very minimal, and we can feel confident that they are behaving correctly.  Changing them would be a huge refactor within nodejs so I feel confident that node will not change them much.

The difficult work is merging upstream changes to something like [node-esm-resolve-implementation.js](https://github.com/TypeStrong/ts-node/blob/main/dist-raw/node-esm-resolve-implementation.js) like in #1167.  For those more complex files, I try to keep them as close as possible to node's own code.

Reply:
Well, it's up to you then. As I mentioned the file size difference is minimal (around ~10Kb), there's no downsides and if we can save a couple of commits in the future due to including the entire primordials, then i'd say worth it. But if you're still against it then I'll revert the change.